### ISSUE TITLE
openMSX refuses to create 2GB harddisk image on 64-bit Windows 10

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1007]
See https://www.msx.org/forum/msx-talk/openmsx/megaflashromsccsd-mapper-emulation-bug?page=1

The user, who is running 64-bit openMSX on a 64-bit Windows 10, with probably an NTFS file system, tried to create a 4GB SD card (hard disk) image with the MegaFlashROM SCC+ SD extension. But openMSX refused:

```
Couldn't create HD image: Files >= 2GB are not supported on 32-bit platforms: C:/Users/SeVe/Documents/openMSX/persistent/MegaFlashROM_SCC+_SD/untitled1/SDcard1.sdc
Uncaught exception: Files >= 2GB are not supported on 32-bit platforms: C:/Users/SeVe/Documents/openMSX/persistent/MegaFlashROM_SCC+_SD/untitled1/SDcard1.sdc
```

Looking at the code in src/file/LocalFile.cc line 220 this is reported when fstat returns EOVERFLOW. However, I can't even find a non-underscored version of fstat in the Windows documentation on MSDN.... So perhaps the root cause lies in there.

**Comment by MBilderbeek** [comment]
How to reproduce: make sure there is no hd.dsk for your machine or extension with a Sunrise IDE (or for the MegaFlashROM SCC+ SD extension: SDcard1.sdc/SDcard2.sdc) and specify a size of 4096 in the hardware config XML file.

**Comment by MBilderbeek** [comment]
Reported to be fixed by the original reporter.

Fixed with commits fd1a0ad0b62ec2e794c28ca2f39b2722efd6d2ce and d1a2f40eda4c2adf75785047525021f95b3135ac - thanks to Max Feingold for support.
